shared/src/main/scala/Context.scala

```diff
@@ -37,7 +37,8 @@ class Context private (
     private val parent: Option[Context] = None,
     val globals: Globals = Globals(),
     val testMode: Boolean = false,
-    val useStack: Boolean = false
+    val useStack: Boolean = false,
+    private val callStack: mut.Stack[VFun] = mut.Stack(),
```

Comment:
I'm not sure we need an entire `callStack` in a single `Context`. If you make a new `Context` for calling a function, it's basically a single stackframe, not an entirely new call stack (and if you make a new `Context` for a loop, it's not even a new stackframe). I'm thinking we could have an `Option[VFun]` that's a `Some` if this `Context` was created for calling a function and `None` otherwise. Thoughts?

Reply:
Well only the top context has functions added/removed from the stack. Maybe the top context could have an extension method. We just need something that's global.

Reply:
Then I think `callStack` should be in `Globals`. There's no need for every `Context` object to have that field if only one `Context` is going to actually use it